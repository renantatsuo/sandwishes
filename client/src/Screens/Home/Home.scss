.home__diy {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  margin-top: 34px;
  height: 500px;
  width: 300px;
  background: url(../../Assets/diy-background.png);

  .home__diy__button {
    font-size: 24px;
    font-weight: bold;
    color: #ffffff;
    text-decoration: none;
    text-align: center;

    padding: 20px;
    background: #e73737;
    border: none;

    cursor: pointer;
    transition: background ease 300ms;

    &:hover,
    &:focus {
      background: #b70606;
      outline: none;
    }
  }
}
