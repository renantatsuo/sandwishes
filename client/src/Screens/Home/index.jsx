import React, { useState, useEffect } from 'react'
import { Link } from 'react-router-dom'

import Burger from '../../Assets/burger.svg'
import MenuItem from '../../Components/MenuItem/index'
import {
  calculateBasePrice,
  fetchSandwiches,
  fetchIngredients
} from '../../helpers'
import './Home.scss'

const mapSandwiches = (sandwiches) => {
  return sandwiches.map((sandwich) => (
    <MenuItem
      desc={sandwich.description}
      key={sandwich.id}
      photo={sandwich.photo}
      price={sandwich.price}
      title={sandwich.name}
      to={`/order/${sandwich.id}`}
    />
  ))
}

const Home = () => {
  const [sandwiches, setSandwiches] = useState([])

  useEffect(() => {
    Promise.all([fetchSandwiches(), fetchIngredients()]).then(
      ([sandwichList, ingredients]) => {
        setSandwiches(
          sandwichList.map((sandwich) => ({
            ...sandwich,
            photo: Burger,
            price: calculateBasePrice(sandwich.recipe, ingredients)
          }))
        )
      }
    )
  }, [])

  return (
    <>
      <div className="home__diy">
        <Link to="/order" className="home__diy__button">
          Monte você mesmo
        </Link>
      </div>
      <div className="home__menu">
        <h1>Escolher no cardápio</h1>
        {mapSandwiches(sandwiches)}
      </div>
    </>
  )
}

export default Home
