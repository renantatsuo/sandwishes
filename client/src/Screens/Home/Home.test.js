import React from 'react'
import { shallow } from 'enzyme'

import Home from './index'
import {
  fetchIngredients,
  fetchSandwiches,
  calculatePrice
} from '../../Helpers'

jest.mock('../../Helpers')

const mockSandwiches = [
  {
    id: 0,
    name: 'Test',
    description: 'Testing',
    recipe: [0, 1, 2]
  }
]

const mockIngredients = [
  { id: 0, name: 'test 1', price: 1.0 },

  { id: 1, name: 'test 2', price: 2.0 },

  { id: 2, name: 'test 3', price: 3.0 }
]

fetchSandwiches.mockResolvedValue(mockSandwiches)
fetchIngredients.mockResolvedValue(mockIngredients)
calculatePrice.mockImplementation(() => 6)
describe('Test Home Screen', () => {
  it('renders with data', () => {
    const app = shallow(<Home />)

    expect(app).toMatchSnapshot()
  })

  it('renders with data', () => {
    const app = shallow(<Home />)
    const appInstance = app.instance()

    appInstance.componentDidMount()

    return fetchSandwiches().then(() =>
      fetchIngredients().then(() => {
        expect(app.state('sandwiches').length).toBe(1)
        expect(app.state('sandwiches')[0].price).toBe(6)
      })
    )
  })
})
