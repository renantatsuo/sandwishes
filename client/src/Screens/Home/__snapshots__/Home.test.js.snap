// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Screen renders with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="home__diy"
        >
          <Link
            className="home__diy__button"
            replace={false}
            to="/order"
          >
            Monte você mesmo
          </Link>
        </div>,
        <div
          className="home__menu"
        >
          <h1>
            Escolher no cardápio
          </h1>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="home__diy__button"
            replace={false}
            to="/order"
          >
            Monte você mesmo
          </Link>,
          "className": "home__diy",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Monte você mesmo",
            "className": "home__diy__button",
            "replace": false,
            "to": "/order",
          },
          "ref": null,
          "rendered": "Monte você mesmo",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Escolher no cardápio
            </h1>,
            Array [],
          ],
          "className": "home__menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Escolher no cardápio",
            },
            "ref": null,
            "rendered": "Escolher no cardápio",
            "type": "h1",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="home__diy"
          >
            <Link
              className="home__diy__button"
              replace={false}
              to="/order"
            >
              Monte você mesmo
            </Link>
          </div>,
          <div
            className="home__menu"
          >
            <h1>
              Escolher no cardápio
            </h1>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="home__diy__button"
              replace={false}
              to="/order"
            >
              Monte você mesmo
            </Link>,
            "className": "home__diy",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Monte você mesmo",
              "className": "home__diy__button",
              "replace": false,
              "to": "/order",
            },
            "ref": null,
            "rendered": "Monte você mesmo",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Escolher no cardápio
              </h1>,
              Array [],
            ],
            "className": "home__menu",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Escolher no cardápio",
              },
              "ref": null,
              "rendered": "Escolher no cardápio",
              "type": "h1",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
